browser/net/brave_network_delegate_base.h

```diff
@@ -20,6 +28,17 @@ namespace net {
 class URLRequest;
 }
 
+namespace brave {
+static const base::NoDestructor<base::flat_set<base::StringPiece>>
```

Comment:
From `no_destructor.h`: "Furthermore, a NoDestructor<T> should never have global scope as that may require a static initializer"

So i'd suggest to make a function that returns this set, and make `NoDestructor` object a static variable in this function.